Tests/TuistHasherTests/GraphContentHasherTests.swift

```diff
@@ -100,6 +99,118 @@ final class GraphContentHasherTests: TuistUnitTestCase {
         .sorted(by: { $0.target.name < $1.target.name })
 
         // Then
-        XCTAssertEqual(hashedTargets, expectedCachableTargets)
+        #expect(hashedTargets == expectedCachableTargets)
+    }
+
+    @Test
+    func test_contentHashes_with_lock_file_in_xcode_project() async throws {
+        try await fileSystem.runInTemporaryDirectory(prefix: "GraphContentHasherTests") { path in
+            // Given
+            let contentHasher = MockContentHashing()
+            given(contentHasher)
+                .hash(path: .any)
+                .willProduce { $0.pathString }
+            let targetContentHasher = MockTargetContentHashing()
+            given(targetContentHasher)
+                .contentHash(
+                    for: .any,
+                    hashedTargets: .any,
+                    hashedPaths: .any,
+                    additionalStrings: .any
+                )
+                .willProduce { graphTarget, _, _, additionalStrings in
+                    TargetContentHash(
+                        hash: graphTarget.target.name + "-" + additionalStrings.joined(separator: "-"),
+                        hashedPaths: [:]
+                    )
+                }
+            let subject = GraphContentHasher(
+                contentHasher: contentHasher,
+                targetContentHasher: targetContentHasher
+            )
+            let lockFilePath = path.appending(
+                components: "App.xcodeproj",
+                "project.xcworkspace",
+                "xcshareddata",
+                "swiftpm",
+                "Package.resolved"
+            )
+            try await fileSystem.makeDirectory(at: lockFilePath.parentDirectory)
+            try await fileSystem.touch(lockFilePath)
+
+            // When
+            let got = try await subject.contentHashes(
+                for: .test(
+                    path: path,
+                    projects: [
+                        path: .test(
+                            targets: [
+                                .test(
+                                    name: "TargetA"
+                                ),
+                            ]
+                        ),
+                    ]
+                ),
+                include: { _ in true },
+                additionalStrings: []
+            )
+
+            // Then
+            #expect(Array(got.values) == ["TargetA-\(lockFilePath)"])
+        }
+    }
+
+    @Test
+    func test_contentHashes_with_lock_file_in_xcode_workspace() async throws {
+        try await fileSystem.runInTemporaryDirectory(prefix: "GraphContentHasherTests") { path in
```

Comment:
I'd be more explicit about what `path` is, otherwise it's a bit hard to understand what it refers to further down.

```suggestion
        try await fileSystem.runInTemporaryDirectory(prefix: "GraphContentHasherTests") { temporaryDirectory in
```